src/providers/battlenet.ts

```diff
@@ -0,0 +1,43 @@
+import { OAuthConfig, OAuthUserConfig } from "."
+
+export interface BattleNetProfile {
+  at_hash: string
+  sub: string
+  aud: string
+  azp: string
+  iss: string
+  exp: number
+  iat: number
+  battle_tag: string
+  jti: string
+}
+
+export default function BattleNet<P extends Record<string, any> = BattleNetProfile>(
+  options: OAuthUserConfig<P> & {
+    /**
+     * Battle.net employs several region-based concepts to manage and deliver game data:
+     * https://develop.battle.net/documentation/guides/regionality-and-apis
+     */
+    region: string
+  }
+): OAuthConfig<P> {
+  const { region } = options
+
+  const base = region.toUpperCase() === "CN"
+    ? "https://www.battlenet.com.cn/oauth"
+    : `https://${region}.battle.net/oauth`
+
+  return {
+    id: "battlenet",
+    name: "Battle.net",
+    type: "oauth",
+    wellKnown: `${base}/.well-known/openid-configuration`,
```

Comment:
```suggestion
  options: OAuthUserConfig<P>
): OAuthConfig<P> {

  return {
    id: "battlenet",
    name: "Battle.net",
    type: "oauth",
    wellKnown: `${options.issuer}/.well-known/openid-configuration`,
```

Let's not introduce a new concept (`region`), to keep providers uniform :+1:

Reply:
looks good, but maybe is better to throw an error instead of having China issuer (or any other region issuer) as a default, wdyt?

Reply:
hmm...
![image](https://user-images.githubusercontent.com/4062731/152663770-fce1b017-9278-423c-994c-e782fd6696a7.png)
I will set one of the regions of global partition as a default